app/controllers/exam_authorization_requests_controller.rb

```diff
@@ -1,36 +1,42 @@
 class ExamAuthorizationRequestsController < ApplicationController
 
-  before_action :verify_registration_opened!, on: [:create, :update]
+  before_action :set_registration!
+  before_action :set_exam!
+  before_action :verify_registration_opened!
+  before_action :verify_registration_in_current_organization!
 
   def create
-    authorization_request = ExamAuthorizationRequest.find_or_create_by! create_authorization_request_params do |it|
-      it.assign_attributes authorization_request_params
-    end
-    current_user.read_notification! authorization_request.exam_registration
+    authorization_request = @registration.request_authorization! current_user, @exam
+    current_user.read_notification! @registration
     flash.notice = I18n.t :exam_authorization_request_created
     redirect_to root_path
   end
 
   def update
-    ExamAuthorizationRequest.update params[:id], authorization_request_params
+    @registration.update_authorization_request_by_id! params[:id], @exam
     flash.notice = I18n.t :exam_authorization_request_saved
     redirect_to root_path
   end
 
   private
 
-  def create_authorization_request_params
-    authorization_request_params.slice :exam_registration_id, :user, :organization
+  def authorization_request_params
+    params.require(:exam_authorization_request).permit(:exam_id, :exam_registration_id)
   end
 
-  def authorization_request_params
-    params
-        .require(:exam_authorization_request).permit(:exam_id, :exam_registration_id)
-        .merge(user: current_user, organization: Organization.current)
+  def set_registration!
+    @registration = ExamRegistration.find(authorization_request_params[:exam_registration_id])
+  end
+
+  def set_exam!
+    @exam = @registration.exams.find(authorization_request_params[:exam_id])
+  end
+
+  def verify_registration_in_current_organization!
+    raise Mumuki::Domain::NotFoundError unless @registration.organization == Organization.current
```

Comment:
This validation is not needed if you set the registration finding the exam registration into the current organization exams registrations instead of `ExamRegistration` model.

```ruby
Organization.current.exam_registrations.find authorization_request_params[:exam_registration_id]
```

Reply:
You are right! Good catch. Could you change that code, please?

Reply:
Done!